### PR TITLE
feat: Progress indicators in Argo CD UI (#4227)

### PR DESCRIPTION
**Pull request by keithchong**
Signed-off-by: Keith Chong <kykchong@redhat.com>

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)). 

Fixes enhancement:
https://github.com/argoproj/argo-cd/issues/4227
UI change only

### CODE REVIEW

File: ui/src/app/applications/components/application-summary/application-summary.tsx [hunk 1]
```diff
@@ -278,21 +280,25 @@ export const ApplicationSummary = (props: {app: models.Application; updateApp: (
...
         if (confirmed) {
+            setChangeSync(true);
             const updatedApp = JSON.parse(JSON.stringify(props.app)) as models.Application;
             if (!updatedApp.spec.syncPolicy) {
                 updatedApp.spec.syncPolicy = {};
             }
             updatedApp.spec.syncPolicy.automated = {prune, selfHeal};
-            props.updateApp(updatedApp);
+            await props.updateApp(updatedApp);
+            setChangeSync(false);
```
Comment (alexmt) [inline ui/src/app/applications/components/application-summary/application-summary.tsx:290]: Please move `setChangeSync(false);` into final block to make sure spinner is removed even if `props.updateApp(updatedApp)` fails

File: ui/src/app/applications/components/application-summary/application-summary.tsx [hunk 2]
```diff
@@ -278,21 +280,25 @@ export const ApplicationSummary = (props: {app: models.Application; updateApp: (
...
         }
     }
 
     async function unsetAutoSync(ctx: {popup: PopupApi}) {
         const confirmed = await ctx.popup.confirm('Disable Auto-Sync?', 'Are you sure you want to disable automated application synchronization');
         if (confirmed) {
+            setChangeSync(true);
             const updatedApp = JSON.parse(JSON.stringify(props.app)) as models.Application;
             updatedApp.spec.syncPolicy.automated = null;
-            props.updateApp(updatedApp);
```
Comment (alexmt) [inline ui/src/app/applications/components/application-summary/application-summary.tsx:295]: Please move `setChangeSync(false);` into final block to make sure spinner is removed even if `props.updateApp(updatedApp)` fails

**Review by alexmt — CHANGES_REQUESTED** [review]
Thank you! Added couple minor comments.

File: ui/src/app/applications/components/application-summary/application-summary.tsx (hunk 2 above)
Comment (keithchong) [reply ui/src/app/applications/components/application-summary/application-summary.tsx:295]: Hi Alex, done.

File: ui/src/app/applications/components/application-summary/application-summary.tsx (hunk 1 above)
Comment (keithchong) [reply ui/src/app/applications/components/application-summary/application-summary.tsx]: Hi Alex, done.

**Review by alexmt — APPROVED** [review]
LGTM

**Pull request merged**